### DISCUSSION TITLE
how can you pass dynamic args to useContractWrite?

### DISCUSSION THREAD

**Discussion by oveddan** [wevm/wagmi#749] [Q&A]
I want to have a single hook for useContractWrite - but be able to pass the args dynamically, for example:

```ts
import deployedContracts from "./contracts/contract.json";


  const { data: signerData } = useSigner();
  const { data, isError, isLoading, writeAsync } = useContractWrite(
    {
      addressOrName: "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512",
      contractInterface: deployedContracts,
      signerOrProvider: signerData,
    },
    "updateToken"
  );

  const callContract = useCallback((tokenId: string) => {
      const args = [tokenId];
      // how can I dynamically pass this args argument?
      writeAsync(args);
  }, []);
```

The above line of writeAsyn(args) doesn't seem to be supported.  How can I pass the args dynamically without setting them in the hook?  For example if I want to have this single hook in my root component, but pass down this `callContract` fn to children that can invoke it with a specific token id.

**Comment by oveddan** [comment]
Looks like I was using an older version of wagmi and this is now possible!

**Reply by arlgnv** [reply]
Can you give an example code of it? I am struggling with it now)

**Reply by jxom** [reply]
Check the docs here: https://wagmi.sh/docs/hooks/useContractWrite#override-config
